### PR TITLE
Prohibitednetworks

### PR DESCRIPTION
**Pull request by hristog121**
This is checking if the IP is public or Private. Can someone check it please

### CODE REVIEW

File: src/checks/prohibeted-networks.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import dns.resolver
+from IPy import IP
+import ipaddress
+
+# Hard coded special IPs that the imports don't cover
+bad_1 = "203.0.113.0"
+bad_2 = "224.0.0.0"
+
+
+def prohibited_chech(ns_server):
```
Comment (KristapsKalnins) [inline src/checks/prohibeted-networks.py]: Fix this typo!

File: src/checks/prohibeted-networks.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            print("[OK]")
+
+
+def print_fail():
+    print("[-] Error ! The IP is not a global IP.")
+    print("[-] TEST FAIL")
+
+
+# Hard code check
+prohibited_chech("198.51.100.0")
```
Comment (KristapsKalnins) [inline src/checks/prohibeted-networks.py]: This one too!

File: src/checks/prohibeted-networks.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    result = dns.resolver.query(ns_server, 'A')
+    for ipval in result:
+        if (ipaddress.ip_address(str(ipval)).is_private or
+                str(ipval) == bad_1 or
+                str(ipval) == bad_2):
+            print_fail()
+
+        else:
+            print("[+] The IP for the for " + ns_server + " is: " + ipval.to_text())
+            print("[OK]")
```
Comment (KristapsKalnins) [inline src/checks/prohibeted-networks.py:32]: Shouldn't your code return a False or True object, not just print the result to the console?

**Review by KristapsKalnins — CHANGES_REQUESTED** [review]
Check my comments.

File: src/checks/prohibeted-networks.py (hunk 3 above)
Comment (hristog121) [reply src/checks/prohibeted-networks.py:32]: No, it should return only the print. In the if statement I have checked if the extracted IP is private, if it is not  that means that the test passes. That's why I give the user a message.

**Review by KristapsKalnins — APPROVED** [review]
Everything looks good! @kasgel should probably have a quick look before merging.

File: src/checks/prohibeted-networks.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# ns_server --> should be taken as an argument from the command line
+# ip_val --> extracted value of the ns server
+import dns.resolver
+from IPy import IP
+import ipaddress
+
+# Hard coded special IPs that the imports don't cover
+bad_1 = "203.0.113.0"
+bad_2 = "224.0.0.0"
+
```
Comment (kasgel) [inline src/checks/prohibeted-networks.py]: It checks for 203.0.113.0 only. On the IANA web page the prohibited network is 203.0.113.0/24. That is, the range 203.0.113.0 - 203.0.113.255.
Same for bad_2

Comment (hristog121) [reply src/checks/prohibeted-networks.py]: OK. I will search for a way to solve that.

**Pull request merged**